### PR TITLE
add gen_string_with_chars & optimize random utilites

### PR DESCRIPTION
**Pull request by TheIronBorn**

### CODE REVIEW

File: src/utils.rs [hunk 1]
```diff
@@ -60,8 +60,10 @@ pub fn dprint<S: Into<String>>(s: S, delay: u64) {
 
 pub fn gen_hex_string(rng: &mut ThreadRng, length: u64) -> String {
     const HEX_CHARS: &[u8] = b"0123456789abcdef";
+    assert!(HEX_CHARS.len().is_power_of_two());
```
Comment (svenstaro) [inline src/utils.rs]: I don't really like the runtime assert to what is a static code hint.

**Comment by TheIronBorn** [comment]
The assert should optimize out.

**Comment by TheIronBorn** [comment]
I could make this general quite easily. Should the set of characters passed in be a `String` or `&[u8]` (via `b”abc”` eg)?

**Comment by TheIronBorn** [comment]
Here's a quick example

**Comment by svenstaro** [comment]
That's great, thanks.

**Pull request merged**